filament/backend/src/vulkan/VulkanContext.cpp

```diff
@@ -275,6 +275,38 @@ void VulkanContext::createLogicalDevice() {
     };
     vmaCreateAllocator(&allocatorInfo, &allocator);
 
+    const uint32_t memTypeBits = UINT32_MAX;
+
+    // Create pools for GPU memory and CPU memory that each have 128 MiB blocks. Crucially, these
+    // pools ignore the "bufferImageGranularity" field in the VkPhysicalDeviceLimits, which means
+    // that all resources allocated from these pools need to have homogeneous linearity. We have
+    // observed that honoring "bufferImageGranularity" can cause the allocator to slow to a crawl.
+    uint32_t memTypeIndex = UINT32_MAX;
+    const VmaAllocationCreateInfo gpuInfo = { .usage = VMA_MEMORY_USAGE_GPU_ONLY };
+    UTILS_UNUSED_IN_RELEASE VkResult res = vmaFindMemoryTypeIndex(allocator, memTypeBits, &gpuInfo,
+            &memTypeIndex);
+    assert_invariant(res == VK_SUCCESS && memTypeIndex != UINT32_MAX);
+    const VmaPoolCreateInfo gpuPoolInfo {
+        .memoryTypeIndex = memTypeIndex,
+        .flags = VMA_POOL_CREATE_IGNORE_BUFFER_IMAGE_GRANULARITY_BIT,
+        .blockSize = 128ull * 1024 * 1024,
```

Comment:
Would it be useful to make this a C++ constant somewhere?

Reply:
Doh, in fact, we have an entire file for stuff like this (VulkanConstants.h).  Fixed.